client/source/js/modules/analysis/optimization-ctrl.js

```diff
@@ -605,8 +600,32 @@ define(['./module', 'angular', 'd3'], function (module, angular, d3) {
     });
 
     $scope.saveOptimization = function () {
-      $http.post('/api/analysis/optimization/save')
-        .success(updateGraphs);
+      var title = 'Save optimization';
+
+      var label = 'New optimization name';
+
+      var options = {
+        description: 'If you would you like to save current objectives and constraints as a new optimization,' +
+          ' please type a new optimization name in a field below. Otherwise just press "Cancel"' +
+          ' and existing optimization "' + $scope.activeOptimization.name + '" will be updated.',
+        acceptButton: 'Yes, save as new',
+        rejectButton: 'No, override current',
+        closeReason: 'overrideoptimization'
+      };
+
+      var doSave = function (name) {
+        $http.post('/api/analysis/optimization/save', { name: name })
+          .success(updateGraphs);
+      };
+
+      modalService.showPrompt(title, label,
+        function (newOptimizationName) {
```

Comment:
Dmitry is using the modal service I've introduced a long ago.
I'm almost got a tear of emotion :D

:trollface: